src/table/components/virtualized-table/__tests__/TableContainer.spec.tsx

```diff
@@ -69,7 +106,90 @@ describe('<TableContainer />', () => {
     const { getByTestId, getByText } = renderTableContainer();
 
     expect(getByTestId('table-container')).toBeVisible();
+    await waitFor(() => expect(getByText('title-0')).toBeVisible()); // A header value
     await waitFor(() => expect(getByText('title-1')).toBeVisible()); // A header value
-    await waitFor(() => expect(getByText('1')).toBeVisible()); // A cell value
+    await waitFor(() => expect(getByText('dimension-c0-r0')).toBeVisible()); // A dimension value
+    await waitFor(() => expect(getByText('dimension-c1-r0')).toBeVisible()); // A dimension value
+    await waitFor(() => expect(getByText('measure-c2-r0')).toBeVisible()); // A measure value
+  });
+
+  describe('selections', () => {
+    it('should be able to select a dimension cell', async () => {
+      const cellText = 'dimension-c0-r0';
+
+      const { getByText, user } = renderTableContainer();
+
+      await waitFor(() => expect(getByText(cellText)).toBeVisible());
+
+      await waitFor(() => expect(getByText(cellText).parentNode).not.toHaveClass('selected'));
+
+      user.click(getByText(cellText));
+
+      await waitFor(() => expect(getByText(cellText).parentNode).toHaveClass('selected'));
```

Comment:
Don't really like the idea of checking a class exists or not. But only other option I could come up with was to add a testid which would add some additional logic just to make the code slight better in terms of testability. Still undecided on this I guess.